junit/src/main/java/cucumber/runtime/junit/JUnitReporter.java

```diff
@@ -50,17 +50,17 @@ public void startExecutionUnit(ExecutionUnitRunner executionUnitRunner, RunNotif
         executionUnitNotifier.fireTestStarted();
     }
 
-    public void finishExecutionUnit() {
+    public void skipExecutionUnitIfIgnored() {
         if (ignoredStep) {
             executionUnitNotifier.fireTestIgnored();
         }
-        executionUnitNotifier.fireTestFinished();
```

Comment:
This was the only fireTestFinished() on the executionUnitNotifier (all others are on the stepNotifier), now the scenario (or in JUnit terms the test suite) will not be finished. The log for the [picocontainer tests](https://travis-ci.org/cucumber/cucumber-jvm/jobs/47766656) visualize this clearly (line 5758).